crates/bevy_app/src/app.rs

```diff
@@ -390,7 +390,9 @@ impl App {
             if let Some(schedule) = schedules.get_mut(&*schedule_label) {
                 schedule.add_system(system);
             } else {
-                panic!("Schedule {schedule_label:?} does not exist.")
+                self
+                    .add_schedule(schedule_label, Schedule::new())
+                    .add_system(system);
```

Comment:
This just adds the system to the app's default schedule, not the newly created one. You'll have to create a new schedule, add the system to it, then add the new schedule to the `schedules` resource. Note that `Schedules::insert` won't work here because of the `Box<dyn ScheduleLabel>` bug, so you'll have to create a new method for adding schedules that takes a `Box<dyn ScheduleLabel>`.

Reply:
> That said, I'd be happy to see a pair of tests for this behavior, one for the single system case and the other for multiple systems.

should these tests be integration tests, in `bevy_app/tests` or it will be ok to place them in [crates/bevy_app/src/app.rs test section](https://github.com/bevyengine/bevy/blob/9c98f8adc35879977fca3a5ea1eadbcb75e4fbe8/crates/bevy_app/src/app.rs#L1016)  ?

Reply:
so, since `App.add_schedule` run `Schedules::insert`, should something like this work?
```rust
let mut schedule = Schedule::new();
schedule.add_system(system);
self.add_schedule(schedule_label, schedule);
              
```

Reply:
That should work. Though I would use `schedules.insert` instead of `self.add_schedule` -- that way you can avoid looking up the `Schedules` resource twice.

Reply:
Unfortunately, this is not working. I added a test and for some reason, it fails if `add_state` is at the end of the chain. Might need help with this...

Reply:
seems like this happening because `add_state` will overwrite `OnEnter` and `OnExit` schedules for all states without system attached

Reply:
should be fixed now

Reply:
It's interesting that `add_state` used to clear the schedule if it already existed. That behavior wasn't documented, so I'm okay with changing it.